cocotb/log.py

```diff
@@ -57,11 +57,14 @@ def __init__(self, name):
 
         # customizations of the defaults
         hdlr = logging.StreamHandler(sys.stdout)
-        want_ansi = os.getenv("COCOTB_ANSI_OUTPUT") and not os.getenv("GUI")
-        if want_ansi is None:
-            want_ansi = sys.stdout.isatty()  # default to ANSI for TTYs
-        else:
-            want_ansi = want_ansi == '1'
+
+        # Highlight with ANSI colors if possible/requested and not running in GUI
+        want_ansi = sys.stdout.isatty()  # default to ANSI for TTYs
+        if os.getenv("COCOTB_ANSI_OUTPUT") is not None:
```

Comment:
This should be checking the value of the string I think, so that passing 0 works

Reply:
I thought of doing this but wanted to keep the logic that was implemented so far.
I'll do this in a separate PR.

Reply:
The original did understand 0 / require it to be set to 1, so I'd argue this is part of gh-809

Reply:
Oh, okay. Would you also allow "yes" and "true" (after lowercasing) to mean``True``?

Reply:
Checking for `'0'` and not `'0'` is enough to restore the old behavior, so that's probably all I'd do in this PR. Does getenv take a default argument? If so, you could set that to `'0'`